### PR TITLE
Fix for repeating workouts over the limit - Issue #7

### PR DESCRIPTION
**Pull request by halfordfan**
When retrieving workouts, the page size needs to stay fixed even when pulling a partial page.

### CODE REVIEW

**Comment by justmedude** [comment]
Quick question, what happens if there are no results on the last page (the total number of workouts is exactly divisible by the limit). Does that page return empty, or is there an error?

**Comment by halfordfan** [comment]
That is a case I have not tested.  This is why I am in management. :-)  I’ll try this today and update as necessary.

**Comment by halfordfan** [comment]
Ok, I think that's going to do it.  1 will be added to the page count if there is a partial page.

File: pylotoncycle/pylotoncycle.py [hunk 1]
```diff
@@ -74,8 +74,7 @@ def GetWorkoutList(self, num_workouts=None):
             num_workouts = self.total_workouts
 
         limit = 100
-        pages = num_workouts // limit
-        rem = num_workouts % limit
+        pages = ( num_workouts // limit ) + ( num_workouts % limit > 0 )
```
Comment (justmedude) [inline pylotoncycle/pylotoncycle.py:77]: Small nit, mostly from the standpoint of readability.
How about keeping the original:
```
        pages = num_workouts // limit
        rem = num_workouts % limit
```
and then we can go:
```
if rem > 0:
    # There are still some remaining workouts, so add another page
    # to capture them all
    pages += 1
```

**Review by justmedude — CHANGES_REQUESTED** [review]
small request
...from one manager to another :)

**Comment by justmedude** [comment]
Btw, really appreciate your contribution!

**Pull request merged**